docs/src/devguide/debugging_the_cython_compiler.rst

```diff
@@ -1,25 +1,34 @@
-Getting started
-^^^^^^^^^^^^^^^
-
-  or: where does that C code come from?
+"Where does that C code come from?"
+===================================
 
 Most often, when you are new to Cython development, you have an idea about the Cython code you want to debug.
 So, looking at the generated C code, your main question will be: »where is that C code generated?«.
 
-Luckily, the Cython compiler has a couple of debug features that you can use
-to pin-point the relevant code sections.  They can be enabled in
-the module `Cython.Compiler.DebugFlags <https://github.com/cython/cython/blob/master/Cython/Compiler/DebugFlags.py>`_.
-Read the comments in that file, enable the relevant debug features
-(usually ``debug_trace_code_generation`` to get started), and then read
-the C code that Cython generates to find out what is going on.
+One simple and useful feature to help you start: by default Cython injects the source
+Python code into the C code as comments in the form:
+
+.. code-block:: c
 
-One trick to identify the origin of *a specific line* of C code is
-to use the Python debugger to stop when that specific line is written by adding
-``if s.find("your line of code") != -1: import pdb; pdb.set_trace()`` to the
-`CCodeWriter.write function in Code.py <https://github.com/cython/cython/blob/b9cecd602878334173aa9f6ed635d48739bfa2b1/Cython/Compiler/Code.py#L1820>`_.
+    /* "source_file.pyx":2
+     * def print_hello():
+     *    print("Hello world")       # <<<<<<<<<<<<<<
+     */
+    
+    __pyx_t_1 = /* some complicated C expression */;
 
-It's usually best to write a test case for the code you want to debug. See the next section on how to do this.
+so searching the generated C code file for the line of Python/Cython code you're interested in can be
+a good place to start.  Some lines may appear in more than one place.  For example, functions appear
+in both the implementation of the function and the place where it's added to the module namespace.
+
+To get more detail, the Cython compiler has a couple of debug features that you
+can use to pin-point the relevant code sections. They can be enabled in
+the module `Cython.Compiler.DebugFlags <https://github.com/cython/cython/blob/master/Cython/Compiler/DebugFlags.py>`_.
+Read the comments in that file, enable the relevant debug features
+(usually ``debug_trace_code_generation`` to get started), and then
+read the C code that Cython generates to find out what is going on.
 
-To get a general idea about the different `code modules <https://github.com/cython/cython/tree/master/Cython/Compiler>_`
-of the Cython compiler, you might want to skip through
-`Josh Kantor's Cython notes <https://web.archive.org/web/20170511222231/https://wstein.org/wiki/attachments/2008(2f)sageseminar(2f)kantor/slides.pdf>`_.
```

Comment:
Just moved to the list of resources in the table of contents. It still looks basically useful to me